### PR TITLE
Ingrediants save use case

### PR DESCRIPTION
**Pull request by Shaffaan**
Hello, there are quite a few changes in this so I will go over them. 
Firstly i refactored the entities, created factories and interfaces and implemented them
Secondly I implemented user and fridge creation use case. I have followed all rules of clean architecture. Now just need to add database functionality to it, which is not difficult to all. This will be the first use case that the user uses, and the controller for this should be injected in the start page
Thirdly I have implemented the enter ingredients use case, it is not complete there is still 1 mirror fix that is needed, but I am not sure how to do it. That is I am not sure how I can get the current user inside the interactor/(how to access the database and how the program will know what the current user is). Will discuss this tomorrow when we meet in person. This method is to be called when user enters an ingredient on the main page of the program.
Still need to do some additional testing. :)
Please let me know if you find errors or if there is stuff I should change

### CODE REVIEW

File: src/main/java/controllers/CreateUserandFridgeController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package controllers;
+
+import useCases.CreateUserAndFridge.CreateUserAndFridgeInputBoundry;
+import useCases.CreateUserAndFridge.CreateUserandFridgeRequestModel;
+import useCases.CreateUserAndFridge.CreateUserandFridgeResponseModel;
+
+public class CreateUserandFridgeController{
```
Comment (EricKarpovits) [inline src/main/java/controllers/CreateUserandFridgeController.java:10]: Please add some doc string comments to this class.

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Override
+    public CreateUserandFridgeResponseModel create(CreateUserandFridgeRequestModel requestModel) {
+        if (Objects.equals(requestModel.getUserName(), "")) {
+            return createUserAndFridgePresenter.prepareFailView("Nothin entered");
+        }
+        else {
+            ArrayList<CommonIngredient> tesst = new ArrayList<>();
+            CommonFridge x = (CommonFridge) fridgeFactory.create(tesst);
```
Comment (EricKarpovits) [inline src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: A more meaningful variable name than "x" might make the code more understandable in the future.

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeResponseModel.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package useCases.CreateUserAndFridge;
+
+import entities.Fridge.CommonFridge;
+import entities.user.CommonUser;
+
+public class CreateUserandFridgeResponseModel {
+    public CommonFridge commonFridge;
```
Comment (EricKarpovits) [inline src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeResponseModel.java]: Please add docstring comments here.

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        this.userFactory = userFactory;
+        this.fridgeFactory = fridgeFactory;
+        this.createUserAndFridgePresenter = createUserAndFridgePresenter;
+
+    }
+
+    @Override
+    public CreateUserandFridgeResponseModel create(CreateUserandFridgeRequestModel requestModel) {
+        if (Objects.equals(requestModel.getUserName(), "")) {
+            return createUserAndFridgePresenter.prepareFailView("Nothin entered");
```
Comment (EricKarpovits) [inline src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: I think "Nothing" might be better.

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import entities.Ingrediant.CommonIngredient;
+import entities.user.CommonUser;
+import entities.user.UserFactory;
+import presenters.CreateUserandFridge.CreateUserAndFridgePresenter;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class CreateUserandFridgeInteractor implements CreateUserAndFridgeInputBoundry {
```
Comment (EricKarpovits) [inline src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: Please add docstring comments.

File: src/main/java/presenters/enteringrediant/UserEnterIngrediantPresenter.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package presenters.enteringrediant;
+
+import entities.Ingrediant.CommonIngredient;
+import useCases.UserEnterIndrediantsInteractor.UserEnterIngrediantResponseModel;
+
+public interface UserEnterIngrediantPresenter {
```
Comment (EricKarpovits) [inline src/main/java/presenters/enteringrediant/UserEnterIngrediantPresenter.java]: Please add docstring comments.

File: src/main/java/presenters/enteringrediant/UserEnterIngrediantFormatter.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package presenters.enteringrediant;
+
+import entities.Ingrediant.CommonIngredient;
+import useCases.UserEnterIndrediantsInteractor.UserEnterIngrediantResponseModel;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class UserEnterIngrediantFormatter implements UserEnterIngrediantPresenter {
+
```
Comment (EricKarpovits) [inline src/main/java/presenters/enteringrediant/UserEnterIngrediantFormatter.java]: Please add docstring comments.

File: src/main/java/presenters/CreateUserandFridge/CreateUserAndFridgePresenter.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package presenters.CreateUserandFridge;
+
+import entities.user.CommonUser;
+import useCases.CreateUserAndFridge.CreateUserandFridgeRequestModel;
+import useCases.CreateUserAndFridge.CreateUserandFridgeResponseModel;
+import useCases.UserEnterIndrediantsInteractor.UserEnterIngrediantResponseModel;
+
+    public interface CreateUserAndFridgePresenter {
```
Comment (EricKarpovits) [inline src/main/java/presenters/CreateUserandFridge/CreateUserAndFridgePresenter.java]: Add some docstring comments here, please.

File: src/main/java/presenters/CreateUserandFridge/CreateUserAndFridgeFormatter.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package presenters.CreateUserandFridge;
+
+import entities.user.CommonUser;
+import useCases.CreateUserAndFridge.CreateUserandFridgeResponseModel;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class CreateUserAndFridgeFormatter implements CreateUserAndFridgePresenter {
+//    @Override
```
Comment (EricKarpovits) [inline src/main/java/presenters/CreateUserandFridge/CreateUserAndFridgeFormatter.java]: Please add docstring comments.

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+                                         createUserAndFridgePresenter) {
+        this.userFactory = userFactory;
+        this.fridgeFactory = fridgeFactory;
+        this.createUserAndFridgePresenter = createUserAndFridgePresenter;
+
+    }
+
+    @Override
+    public CreateUserandFridgeResponseModel create(CreateUserandFridgeRequestModel requestModel) {
+        if (Objects.equals(requestModel.getUserName(), "")) {
```
Comment (EricKarpovits) [inline src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: Potentially .isEmpty() method might make the code more cleaner - not sure.

File: src/main/java/useCases/UserEnterIndrediantsInteractor/UserEnterIngredientRequestModel.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
+package useCases.UserEnterIndrediantsInteractor;
+
+public class UserEnterIngredientRequestModel {
+
+    public String Ingredient;
+
```
Comment (EricKarpovits) [inline src/main/java/useCases/UserEnterIndrediantsInteractor/UserEnterIngredientRequestModel.java]: Please make instance variables camelCase.

**Review by EricKarpovits — CHANGES_REQUESTED** [review]

**Comment by EricKarpovits** [comment]
Great work @Shaffaan please make sure to add docstring comments to every class.

File: src/main/java/controllers/CreateUserandFridgeController.java (hunk 1 above)
Comment (Shaffaan) [reply src/main/java/controllers/CreateUserandFridgeController.java:10]: Added Extra docstrings

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java (hunk 2 above)
Comment (Shaffaan) [reply src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: Changed the name to curr_fridge

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java (hunk 10 above)
Comment (Shaffaan) [reply src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: Changed it

File: src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java (hunk 4 above)
Comment (Shaffaan) [reply src/main/java/useCases/CreateUserAndFridge/CreateUserandFridgeInteractor.java]: Done changed it

File: src/main/java/presenters/enteringrediant/UserEnterIngrediantFormatter.java (hunk 7 above)
Comment (Shaffaan) [reply src/main/java/presenters/enteringrediant/UserEnterIngrediantFormatter.java]: Added

File: src/main/java/presenters/CreateUserandFridge/CreateUserAndFridgePresenter.java (hunk 8 above)
Comment (Shaffaan) [reply src/main/java/presenters/CreateUserandFridge/CreateUserAndFridgePresenter.java]: Added

File: src/main/java/entities/Ingrediant/CommonIngredient.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    public CommonIngredient(String name) {
+        this.name = name;
+    }
+
+    /**
+     *
+     * @return The name of the ingredient
+     */
+    public String getName() {
+        return name;
```
Comment (fishyvishy) [inline src/main/java/entities/Ingrediant/CommonIngredient.java]: this.name

File: src/main/java/entities/user/CommonUser.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
+package entities.user;
+
+import entities.Fridge.CommonFridge;
+
+/**
+ * Class CommonUser
+ */
+public class CommonUser implements User {
+    String name;
```
Comment (fishyvishy) [inline src/main/java/entities/user/CommonUser.java:9]: add access modifiers

File: src/main/java/entities/user/User.java [hunk 14]
```diff
@@ -0,0 +1,13 @@
+package entities.user;
+
+import entities.Fridge.CommonFridge;
+
+/**
+ * Interface user
+ * implemented by CommonUser class
+ */
+public interface User {
+    boolean ingredientIsValid();
```
Comment (fishyvishy) [inline src/main/java/entities/user/User.java:10]: access modifiers

File: src/main/java/entities/Ingrediant/CommonIngredient.java (hunk 12 above)
Comment (Shaffaan) [reply src/main/java/entities/Ingrediant/CommonIngredient.java]: ok will change it

**Review by fishyvishy — APPROVED** [review]
Looks good. Will work on patching it with the start page UI.

File: src/main/java/entities/user/CommonUser.java (hunk 13 above)
Comment (Shaffaan) [reply src/main/java/entities/user/CommonUser.java:9]: done

File: src/main/java/presenters/enteringrediant/UserEnterIngredientPresenter.java [hunk 15]
```diff
@@ -0,0 +1,10 @@
+package presenters.enteringrediant;
+
+import useCases.UserEnterIndrediantsInteractor.UserEnterIngredientResponseModel;
+
+public interface UserEnterIngredientPresenter {
```
Comment (EricKarpovits) [inline src/main/java/presenters/enteringrediant/UserEnterIngredientPresenter.java]: Maybe a docstring here

**Review by EricKarpovits — APPROVED** [review]
Great work @Shaffaan. One mior comment.

File: src/main/java/entities/Fridge/CommonFridgeFactory.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
+package entities.Fridge;
+
+import entities.Ingrediant.CommonIngredient;
+
+import java.util.ArrayList;
+
+/**
+ * Class CommonFridgeFactory implements the FridgeFactory interface
+ */
+public class CommonFridgeFactory implements FridgeFactory {
```
Comment (EricKarpovits) [inline src/main/java/entities/Fridge/CommonFridgeFactory.java]: Please implement Serializable.

**Review by rickychen175 — APPROVED** [review]
Looks Good

File: src/main/java/presenters/enteringrediant/UserEnterIngredientPresenter.java (hunk 15 above)
Comment (Shaffaan) [reply src/main/java/presenters/enteringrediant/UserEnterIngredientPresenter.java]: added

**Pull request merged**